spot-cluster-sizing.md

```diff
@@ -0,0 +1,80 @@
+Spot Cluster Sizing
+===================================
+
+## Recommended cluster configuration
+
+The recommended spot cluster configuration uses all of the data available to
+Kubecost to compute a "resizing" of your cluster's nodes into a set of on-demand
+(standard) nodes `O` and a set of spot (preemptible) nodes `S`. This
+configuration is produced from applying a scheduling heuristic to the usage data
+for all of your workloads. This recommendation offers a more accurate picture of
+the savings possible from implementing spot nodes because nodes are what the
+cost of a cluster is made up of; once `O` and `S` have been determined, the
+savings are the current cost of your nodes minus the estimated cost of `O` and
+`S`.
+
+
+### Implementing the recommended configuration
+
+The recommended configuration assumes that all workloads considered spot-ready
+by the [Spot Checklist](./spot-checklist.md) will be schedulable on spot nodes
+and that workloads considered not spot-ready will only be schedulable on
+on-demand nodes. Kubernetes has [taints and
+tolerations](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
+for achieving this behavior. Cloud providers usually have guides for using spot
+nodes with taints and tolerations in your managed cluster:
+
+- AWS (EKS) <https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html#managed-node-group-capacity-types>
+- GCP (GKE) <https://cloud.google.com/kubernetes-engine/docs/how-to/preemptible-vms>
+- Azure (AKS) <https://docs.microsoft.com/en-us/azure/aks/spot-node-pool>
+
+
+Different cloud providers have different guarantees on shutdown windows and automatic
+draining of spot nodes that are about to be removed. Consult your provider&rsquo;s
+documentation before introducing spot nodes to your cluster.
+
+Kubecost marking a workload as spot ready is not a guarantee. A domain expert should
+always carefully consider the workload before approving it to run on spot nodes.
+
+### How the recommended cluster configuration is determined
+
+Determing `O` and `S` is achieved by first partitioning all workloads on the
+cluster (based on the results of the Checklist) into sets:
+spot-ready workloads `R` and non-spot-ready workloads `N`. Kubecost consults its
```

Comment:
Is there a way to set workloads to `R` or `N` manually, so that the command output is adjusted accordingly, or would the user just hack and slash at the command as they saw fit?

Reply:
This is an excellent question. Right now, `R` and `N` can only come from the spot checklist which is based on the cluster state and the checklist's logic. Users can manually move a workload from `N` to `R` by using the annotation overrides, but this is clumsy as an experimental tool because it actually modifies cluster state. What I would like to implement, if the spot features get traction, is something on the UI to "hypothetically" mark workloads as spot ready (i.e. move them from `N` to `R` for the sake of experiment) and get a live-updated cluster sizing recommendation.